plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py

```diff
@@ -160,6 +160,8 @@ def sweep(self, arguments: List[str]) -> None:
             "tpe": "optuna.samplers.TPESampler",
             "random": "optuna.samplers.RandomSampler",
             "cmaes": "optuna.samplers.CmaEsSampler",
+            "nsgaii": "optuna.samplers.NSGAIISampler",
+            "motpe": "optuna.samplers.MOTPESampler",
```

Comment:
Some of the samplers are taking parameters:
e.g:
https://optuna.readthedocs.io/en/stable/reference/generated/optuna.samplers.CmaEsSampler.html#

```python
CmaEsSampler(x0=None, sigma0=None, n_startup_trials=1, independent_sampler=None, warn_independent_sampling=True, seed=None, *, consider_pruned_trials=False, restart_strategy=None, inc_popsize=2)[
```

We can have a followup PR to expose those to the user.
If we are okay depending on Hydra 1.1 features (not yet released), we should look at using recursive defaults and having a netsted config group for the samplers.

That would be an interest use case to try out the new capabilities coming in Hydra 1.1.

Reply:
> We can have a followup PR to expose those to the user.
> If we are okay depending on Hydra 1.1 features (not yet released), we should look at using recursive defaults and having a netsted config group for the samplers.

Thank you for letting me know. I agree with you.

I think Optuna users want to specify arguments of samplers. For example, [`population_size`](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.samplers.NSGAIISampler.html#optuna.samplers.NSGAIISampler) of `NSGAIISampler` should be changed depending on `n_trials` because it will be identical to `RandomSampler` if the `population_size` is larger than `n_trials`.

In addition to that, we have still a few users according to the [pypistats](https://pypistats.org/packages/hydra-optuna-sweeper), and I think we need to attract new users by adding new functionalities.

Reply:
> In addition to that, we have still a few users according to the pypistats, and I think we need to attract new users by adding new functionalities.

I think it will mostly take awareness.
Not many people are using HPO plugins with Hydra in general, so I am not surprised that this new HPO plugin is also not getting a lot of usage.